### PR TITLE
Optimizes includes

### PR DESCRIPTION
**Pull request by balazsracz**
Reduces the compilation footprint by optimizing which files include which other files.
This enables writing code that compiles faster.

### CODE REVIEW

File: src/openlcb/Convert.hxx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ * @param error_message will hold all remaining bytes that came with the error
+ * message.
+ */
+extern void buffer_to_error(const Payload& payload, uint16_t* error_code, uint16_t* mti, string* error_message);
+
+/** A global class / variable for empty or not-yet-initialized payloads. */
+extern string EMPTY_PAYLOAD;
+
+/// @return the high 4 bytes of a node ID. @param id is the node ID.
+inline unsigned node_high(NodeID id) {
```
Comment (atanisoft) [inline src/openlcb/Convert.hxx:118]: formatting

File: src/openlcb/Convert.hxx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    return id >> 32;
+}
+/// @return the low 4 bytes of a node ID. @param id is the node ID.
+inline unsigned node_low(NodeID id) {
```
Comment (atanisoft) [inline src/openlcb/Convert.hxx:122]: formatting

File: src/openlcb/Convert.hxx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+ * @returns the node ID represented by the first 6 bytes of d.
+ */
+extern NodeID data_to_node_id(const void* d);
+
+/** Converts an Event ID to a Payload suitable to be sent as an event report. */
+extern Payload eventid_to_buffer(uint64_t eventid);
+
+/** Takes 8 bytes (big-endian) from *data, and returns the event id they
+ * represent. */
+inline uint64_t data_to_eventid(const void* data) {
```
Comment (atanisoft) [inline src/openlcb/Convert.hxx:80]: formatting

File: src/openlcb/Convert.hxx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+#ifndef _OPENLCB_CONVERT_HXX_
+#define _OPENLCB_CONVERT_HXX_
+
+#include <string.h>
+#include <endian.h>
+
+#include "openlcb/Defs.hxx"
+
+namespace openlcb {
```
Comment (atanisoft) [inline src/openlcb/Convert.hxx:43]: formatting

File: src/openlcb/If.hxx [hunk 5]
```diff
@@ -40,6 +40,7 @@
 
 #include "openlcb/Node.hxx"
 #include "openlcb/Defs.hxx"
+#include "openlcb/Convert.hxx"
```
Comment (atanisoft) [inline src/openlcb/If.hxx:43]: should this be before Defs?

File: src/openlcb/TractionThrottleInterface.hxx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+ */
+
+#ifndef _OPENLCB_TRACTIONTHROTTLEINTERFACE_HXX_
+#define _OPENLCB_TRACTIONTHROTTLEINTERFACE_HXX_
+
+#include "openlcb/Defs.hxx"
+#include "executor/CallableFlow.hxx"
+#include "openlcb/TrainInterface.hxx"
+
+namespace openlcb {
```
Comment (atanisoft) [inline src/openlcb/TractionThrottleInterface.hxx:42]: formatting

File: src/openlcb/TrainInterface.hxx [hunk 7]
```diff
@@ -35,11 +35,15 @@
 #ifndef _OPENLCB_TRAININTERFACE_HXX_
 #define _OPENLCB_TRAININTERFACE_HXX_
 
-#include "openlcb/TractionDefs.hxx"
+#include "openlcb/Velocity.hxx"
 #include "dcc/Defs.hxx"
 
 namespace openlcb {
```
Comment (atanisoft) [inline src/openlcb/TrainInterface.hxx:41]: formatting

File: src/openlcb/Velocity.hxx [hunk 8]
```diff
@@ -486,6 +486,12 @@ private:
     }
 };
 
+/** @returns NAN as speed. */
+inline Velocity nan_to_speed() {
```
Comment (atanisoft) [inline src/openlcb/Velocity.hxx:490]: formatting

File: src/openlcb/Convert.hxx (hunk 1 above)
Comment (balazsracz) [reply src/openlcb/Convert.hxx]: fixed all of these

**Pull request merged**